Demo/Demo/ViewModels/CardVaultViewModel.swift

```diff
@@ -0,0 +1,109 @@
+import UIKit
+import CardPayments
+import CorePayments
+
+class CardVaultViewModel: ObservableObject, CardVaultDelegate {
+
+    @Published var state = CardVaultState()
+
+    func getSetupToken(
+        customerID: String? = nil,
+        selectedMerchantIntegration: MerchantIntegration
+    ) async throws {
+        do {
+            DispatchQueue.main.async {
+                self.state.setupTokenResponse = .loading
+            }
+            let setupTokenResult = try await DemoMerchantAPI.sharedService.getSetupToken(
+                customerID: customerID,
+                selectedMerchantIntegration: selectedMerchantIntegration
+            )
+            DispatchQueue.main.async {
+                self.state.setupTokenResponse = .loaded(setupTokenResult)
+            }
+        } catch {
+            DispatchQueue.main.async {
+                self.state.setupTokenResponse = .error(message: error.localizedDescription)
+            }
+            throw error
+        }
+    }
+
+    func resetState() {
+        state = CardVaultState()
+    }
+
+    func getPaymentToken(
+        setupToken: String,
+        selectedMerchantIntegration: MerchantIntegration
+    ) async throws {
+        do {
+            DispatchQueue.main.async {
+                self.state.paymentTokenResponse = .loading
+            }
+            let paymentTokenResult = try await DemoMerchantAPI.sharedService.getPaymentToken(
+                setupToken: setupToken,
+                selectedMerchantIntegration: selectedMerchantIntegration
+            )
+            DispatchQueue.main.async {
+                self.state.paymentTokenResponse = .loaded(paymentTokenResult)
+            }
+        } catch {
+            DispatchQueue.main.async {
+                self.state.paymentTokenResponse = .error(message: error.localizedDescription)
+            }
+            throw error
+        }
+    }
+
+    func vault(
+        config: CoreConfig,
+        card: Card,
+        setupToken: String
+    ) async {
```

Comment:
We can probably single line this and be under our linter line length

Reply:
addressed in commit 121c4eadb22995e34dbf68e2aee497baa92e2249